File: 02. C# Part2/06. StringsАndTextProcessing-Homework/03. CorrectBrackets/CorrectBrackets.cs
using System;
using System.Linq;
using System.Collections.Generic;

//Write a program to check if in a given expression the brackets are put correctly.

    class CorrectBrackets
    {
        private static readonly char[] OpenParentheses = { '(', '[', '{' };
        private static readonly char[] CloseParentheses = { ')', ']', '}' };
        static void Main()
        {
            Console.Write("Enter an expression with brackets: ");
            string expression = Console.ReadLine();

            CheckExpression(expression);
            Console.WriteLine(CheckExpression(expression)
                ? "The brackets in your expression are put correctly."
                : "The brackets in your expression are put incorrectly or are missing.");
        }

        private static bool CheckExpression(string input)
        {
            Stack<int> parentheses = new Stack<int>();

            foreach (char chr in input)
            {
                int index;

                if ((index = Array.IndexOf(OpenParentheses, chr)) != -1)
                {
                    parentheses.Push(index); 
                }
                else if ((index = Array.IndexOf(CloseParentheses, chr)) != -1)
                {
                    if (parentheses.Count == 0 || parentheses.Pop() != index) return false;
                }
            }
            return parentheses.Count == 0;
        }
        }
    

File: 06. JavaScript Basics/08. Strings-Homework/08. Strings-Homework/02. CorrectBrackets/CorrectBrackets.js
//Write a JavaScript function to check if in a given expression the brackets are put correctly.

function isCorrectExpression(expression) {
    var brackets = 0;

    for (var i = 0; i < expression.length; i++) {
        if (expression[i] === '(') {
            brackets++;
        }
        else if (expression[i] === ')') {
            brackets--;
        }
    }
    jsConsole.writeLine(!brackets);
}

isCorrectExpression('(3+4) + (3-4)');
isCorrectExpression(')3+4) + (3-4)');